test/components/views/rooms/RoomListHeader-test.tsx

```diff
@@ -28,12 +30,97 @@ import DMRoomMap from "../../../../src/utils/DMRoomMap";
 import MatrixClientContext from "../../../../src/contexts/MatrixClientContext";
 import SettingsStore from "../../../../src/settings/SettingsStore";
 import { SettingLevel } from "../../../../src/settings/SettingLevel";
+import { shouldShowComponent } from '../../../../src/customisations/helpers/UIComponents';
+import { UIComponent } from '../../../../src/settings/UIFeature';
+
+jest.mock('../../../../src/customisations/helpers/UIComponents', () => ({
+    shouldShowComponent: jest.fn(),
+}));
+
+const blockUIComponent = component => {
+    mocked(shouldShowComponent).mockImplementation(feature => feature !== component);
+};
+
+const setupSpace = (client) => {
+    const testSpace = mkSpace(client, "!space:server");
+    testSpace.name = "Test Space";
+    client.getRoom = () => testSpace;
+    return testSpace;
+};
+
+const setupMainMenu = async (client, testSpace) => {
+    const getUserIdForRoomId = jest.fn();
+    const getDMRoomsForUserId = jest.fn();
+    // @ts-ignore
+    DMRoomMap.sharedInstance = { getUserIdForRoomId, getDMRoomsForUserId };
+
+    await testUtils.setupAsyncStoreWithClient(SpaceStore.instance, client);
+    act(() => {
+        SpaceStore.instance.setActiveSpace(testSpace.roomId);
+    });
+
+    const wrapper = mount(<MatrixClientContext.Provider value={client}>
+        <RoomListHeader />
+    </MatrixClientContext.Provider>);
```

Comment:
You can use the `wrapInMatrixClientContext` function from test/test-utils to do this - see test/components/structures/VideoRoomView-test.tsx for an example

Reply:
I would need to inject my own client in `wrapInMatrixClientContext` as this function holds its own internal client in order to make this work.

Perhaps we could add another function : 
```
export function wrapInMatrixClientContextWithClient<T>(WrappedComponent: ComponentType<T>, client: MatrixClient): ComponentType<WrapperProps<T>> {
    class Wrapper extends React.Component<WrapperProps<T>> {
        _matrixClient: MatrixClient;
        constructor(props) {
            super(props);

            this._matrixClient = client;
        }

        render() {
            return <MatrixClientContext.Provider value={this._matrixClient}>
                <WrappedComponent ref={this.props.wrappedRef} {...this.props} />
            </MatrixClientContext.Provider>;
        }
    }
    return Wrapper;
}
```

Also I might not use the correct approach here (it feels to me that we have not the same issue in VideoRoomView-test.tsx). Please do hesitate to guide me through this if you think there is a better approach. We would be very grateful!

Reply:
The client that `wrapInMatrixClientContext` uses is the one returned by `MatrixClientPeg.get()`, which can be set in tests by using `stubClient` instead of just `createTestClient`:

```diff
- client = createTestClient();
+ stubClient();
+ client = MatrixClientPeg.get();
```

If you do that, `wrapInMatrixClientContext` will automatically pick up the correct client each time it's mounted.

Reply:
thank you